src/components/dialogView/index.js

```diff
@@ -0,0 +1,221 @@
+    /**
+     * An alternative divider style
+     */
+    dividerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative style surrounding the whole header
+     */
+    headerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative style surrounding the title, message and the renderRightOfTitle
+     */
+    titleOuterContainerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative style surrounding both title and message
+     */
+    titleInnerContainerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative style for the title
+     */
+    titleStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative style for the message
+     */
+    messageStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * An alternative container style for the DialogView
+     */
+    containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+  };
+
+  renderKnob = () => {
+    const {knobStyle} = this.getThemeProps();
+    return <View style={[styles.knob, knobStyle]} />;
+  };
+
+  hasTitle = () => {
+    const {title, message, renderRightOfTitle} = this.getThemeProps();
+    return !_.isUndefined(title) || !_.isUndefined(message) || !_.isUndefined(renderRightOfTitle);
+  };
+
+  renderTitle = () => {
+    const {
+      testID,
+      title,
+      message,
+      renderRightOfTitle,
+      titleOuterContainerStyle,
+      titleInnerContainerStyle,
+      titleStyle,
+      messageStyle,
+    } = this.getThemeProps();
+
+    return (
+      <View style={[styles.titleOuterContainer, titleOuterContainerStyle]}>
+        <View style={[styles.titleInnerContainer, titleInnerContainerStyle]}>
+          <Text testID={`${testID}.title`} style={[styles.title, titleStyle]}>
+            {title}
+          </Text>
+          {message && (
+            <Text testID={`${testID}.message`} style={[styles.message, messageStyle]}>
+              {message}
+            </Text>
+          )}
+        </View>
+        {renderRightOfTitle && renderRightOfTitle()}
+      </View>
+    );
+  };
+
+  renderDivider = () => {
+    const {dividerStyle} = this.getThemeProps();
+    return <View style={[styles.divider, dividerStyle]} />;
+  };
+
+  renderHeader = () => {
+    const {headerStyle, showKnob, renderBelowTitle, showDivider} = this.getThemeProps();
+    const hasTitle = this.hasTitle();
+
+    return (
+      <View style={[styles.header, headerStyle]}>
+        {showKnob && this.renderKnob()}
+        {hasTitle && this.renderTitle()}
+        {renderBelowTitle && renderBelowTitle()}
+        {showDivider && this.renderDivider()}
+      </View>
+    );
+  };
+
+  hasHeader = () => {
```

Comment:
let's try to simplify the API for the `DialogView` since it's a public component and move the complex UI logic to the private lib. 
So, in my opinion, we should have basic props for 
- knob
- title, titleStyle
- subtitle, subtitleStyle
- renderHeader (custom header)

and we can avoid too specific props like: `titleOuterContainerStyle`, `titleInnerContainerStyle`, `renderRightOfTitle`, `renderBelowTitle`

Reply:
Done